File: python/raindrops/raindrops.py
def raindrops(number: int) -> str:
    subs = ((3, 'Pling'), (5, 'Plang'), (7, 'Plong'))
    preformat = [plonger for divisor, plonger in subs if number % divisor == 0]
    if not preformat:
        return str(number)
    return ''.join(preformat)

File: ruby/raindrops/raindrops.rb
# frozen_string_literal: true

module Raindrops
  refine Integer do
    def divisible?(divisor)
      modulo(divisor).zero?
    end
  end

  using Raindrops

  SUBS = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }.freeze

  def self.convert(number)
    format = SUBS.map { |divisor, sub| sub if number.divisible? divisor }.join

    return format unless format.empty?

    number.to_s
  end
end
